platforms/hyperledger-besu/charts/node_orion/templates/deployment.yaml

```diff
@@ -0,0 +1,200 @@
+        service.rpc: {{ .Values.node.name }}
+    spec:
+      serviceAccountName: {{ .Values.vault.serviceaccountname }}
+      imagePullSecrets:
+          - name: {{ .Values.node.imagePullSecret }}
+      volumes:
+        - name: certificates
+          emptyDir:
+            medium: Memory
+        - name: {{ .Values.node.name }}-genesis-volume
+          configMap:
+            name: genesis-{{ .Values.node.name }}
+            items:
+              - key: genesis.json.base64
+                path: genesis.json.base64
+      initContainers:
+      - name: certificates-init
+        image: {{ .Values.images.alpineutils }}
+        imagePullPolicy: Always
+        env:
+        - name: VAULT_ADDR
+          value: {{ .Values.vault.address }}
+        - name: VAULT_SECRET_PREFIX
+          value: {{ .Values.vault.secretprefix }}
+        - name: KUBERNETES_AUTH_PATH
+          value: {{ .Values.vault.authpath }}
+        - name: VAULT_APP_ROLE
+          value: {{ .Values.vault.role }}
+        - name: MOUNT_PATH
+          value: "/secret"
+        command: ["sh", "-c"]
+        args:
+        - |-
+          #!/usr/bin/env sh
+          
+          validateVaultResponse () {
+          if echo ${2} | grep "errors"; then
+           echo "ERROR: unable to retrieve ${1}: ${2}"
+           exit 1
+           fi
+          }
+          KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+          echo "Getting secrets from Vault Server: ${VAULT_ADDR}"
+          #Login to Vault and to get an approle token
+          VAULT_CLIENT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login \
+            -H "Content-Type: application/json" \
+            -d '{"role":"'"${VAULT_APP_ROLE}"'","jwt":"'"${KUBE_SA_TOKEN}"'"}' | \
+            jq -r 'if .errors then . else .auth.client_token end')
+          validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
+
+          mkdir -p ${MOUNT_PATH}
+
+          vault_secret_key="${VAULT_SECRET_PREFIX}/{{ .Values.vault.keyname }}"
+          echo "Getting 3 keys from $vault_secret_key"
+          LOOKUP_SECRET_RESPONSE=$(curl -sS \
+            --header "X-Vault-Token:${VAULT_CLIENT_TOKEN}" \
+            ${VAULT_ADDR}/v1/${vault_secret_key} | \
+            jq -r 'if .errors then . else . end')
+          validateVaultResponse "secret (${vault_secret_key})" "${LOOKUP_SECRET_RESPONSE}" 
+          nodekey=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data["key"]')                 
+             
+          echo "${nodekey}" > ${MOUNT_PATH}/nodekey
+
+        volumeMounts:
+        - name: certificates
+          mountPath: /secret
+      - name: besu-genesis-init-container
+        image: "{{ .Values.images.node }}"
+        command: [ "sh" ]
+        args:
+        - "-cx"
+        - "mkdir -p $BESU_GENESIS_DIR;
+          cat {{ .Values.node.mountPath }}/genesis/genesis.json.base64 | base64 -d > $BESU_GENESIS_DIR/genesis.json;
```

Comment:
Don't we run any genesis command here?

Reply:
No

Reply:
We use this to mount fetch and mount genesis data just like we do for crypto.

Reply:
Then we do not need this container, it can be mounted directly on the main container

Reply:
Sure